Client/Frontend/Home/RecentlyClosedTabsPanel.swift

```diff
@@ -117,16 +116,16 @@ class RecentlyClosedTabsPanelSiteTableViewController: SiteTableViewController {
         let tab = recentlyClosedTabs[indexPath.row]
         let displayURL = tab.url.displayURL ?? tab.url
         twoLineCell.setLines(tab.title, detailText: displayURL.absoluteDisplayString)
-        let site: Favicon? = (tab.faviconURL != nil) ? Favicon(url: tab.faviconURL!, type: .guess) : nil
+        let site: Favicon? = tab.faviconURL.flatMap { Favicon(url: $0, type: .guess) }
```

Comment:
I think this is a case where using flatmap is a lot clearer (and more swifty) then what we were doing before. It might be a bit weird at first but I think in places like URLExtensions it has helped clean up a lot of the code.  

`return normalizedHost.flatMap { $0 + self.path }` 

instead of 
```
if let normalizedHost = self.normalizedHost {
    return normalizedHost + self.path
}
return nil
```